### PR TITLE
Preparation for ESAPI 2.5.3.0 release

### PR DESCRIPTION
**Pull request by kwwall**
* Update to AntiSamy 1.7.4
* Javadoc fixes / updates / improvements
* Revise ESAPI's deprecation policy
* Change ESAPI's default key size for digital signatures from 1024-bits to 2048-bits.
* Add 'noopenerAndNoreferrerAnchors' AntiSamy directive for anchor tags.
* Deprecate both `Validator.isValidSafeHTML` interfaces; to be removed in 1 year after ESAPI 2.5.3.0 release.
* Revise / reorganize JUnit tests using deprecated `isValidSafeHTML` methods.

### CODE REVIEW

**Review by xeno6696 — APPROVED** [review]
Straightforward enough.

**Comment by xeno6696** [comment]
Sorry I thought I was just approving the first commit.  Walking through the changes today.

File: src/main/java/org/owasp/esapi/reference/DefaultSecurityConfiguration.java
Comment (xeno6696) [inline src/main/java/org/owasp/esapi/reference/DefaultSecurityConfiguration.java:1079]: Worth asking why we don't just default to 4096?

File: src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleCleanTest.java
Comment (xeno6696) [inline src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleCleanTest.java:320]: ROFL!!!!

"badVoodoo"

Comment (xeno6696) [inline src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleCleanTest.java:320]: Yeah I'm surprised we didn't already test for this.

File: src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleCleanTest.java
Comment (xeno6696) [inline src/test/java/org/owasp/esapi/reference/validation/HTMLValidationRuleCleanTest.java:440]: Surprised we didn't already have this one either...

**Comment by xeno6696** [comment]
I approve, I will wait until EOB for @jeremiahjstacey to take a pass, but this seems mostly routine with just some new tests.

File: src/main/java/org/owasp/esapi/reference/DefaultSecurityConfiguration.java
Comment (kwwall) [reply src/main/java/org/owasp/esapi/reference/DefaultSecurityConfiguration.java:1079]: We probably should switch to using EdDSA or ECDSA if EdDSA is not supported in the standard JDK 8, but for now (and until 2030), I think 2048-bits is considered fine by NIST and I wasn't sure of the performance impact of bumping it to 4096 (it wouldn't be noticeable unless someone was doing a lot of them in a tight loop).  So, I figure if they want it better, let them tweak ESAPI.properties. That's one reason that even though I'd been aware of this for several years, I wasn't overly concerned about it. I pretty much only did it because of a GitHub issue someone opened.

Maybe not a great answer, but at this point, I don't really want to go back and change it.

**Comment by kwwall** [comment]
Okay. I will wait until 7pm ET today until I merge this to give @jeremiahjstacey the opportunity to approve this PR. Of it he doesn't intend reviewing this PR, until he communicates that via a comment in this PR or an email to me.

**Comment by jeremiahjstacey** [comment]
All looks good to me.  Sorry for the delay. 
I appreciate the smaller scoped tests and all the accompanying comments.

**Comment by kwwall** [comment]
@jeremiahjstacey  wrote:
> All looks good to me. Sorry for the delay. I appreciate the smaller scoped tests and all the accompanying comments.

No worries. It's all good. Just want to get this out the door while I'm on PTO / Holiday.

**Pull request merged**
